### ISSUE TITLE
Get.generalDialog sets showGeneralDialog's optional parameters to null by default, resulting in errors and omission of default values.

### ISSUE THREAD

**Issue by MaskyS** [jonataslaw/get#130]
**Describe the bug**
The optional parameters of `generalDialog` are not so optional: Omitting `barrierDismissible` or `transitionDuration` will result in build errors. This is because default values of `showGeneralDialog`'s optional parameters are being set to null by `Get`. Overall, `Get.generalDialog`'s current behavior does not match the "native" experience, both on the dev and UI side (e.g. `barrierColor` set to null) .

**Flutter Version:**
1.17.1 (Stable, May Release)

**Get Version:**
`get: ^2.4.0`

**Describe on which device you found the bug:**
It should be reproducible on all devices. Mine was a Pixel 2XL emulator.

**Minimal reproduce code**
```
await Get.generalDialog(pageBuilder: (context, __, ___) => AlertDialog());
```

P.S. Thanks for this awesome library! It makes separation of UI/logic effortless when paired up with other packages.

**Comment by jonataslaw** [comment]
> **Describe the bug**
> The optional parameters of `generalDialog` are not so optional: Omitting `barrierDismissible` or `transitionDuration` will result in build errors. This is because default values of `showGeneralDialog`'s optional parameters are being set to null by `Get`. Overall, `Get.generalDialog`'s current behavior does not match the "native" experience, both on the dev and UI side (e.g. `barrierColor` set to null) .
> 
> **Flutter Version:**
> 1.17.1 (Stable, May Release)
> 
> **Get Version:**
> `get: ^2.4.0`
> 
> **Describe on which device you found the bug:**
> It should be reproducible on all devices. Mine was a Pixel 2XL emulator.
> 
> **Minimal reproduce code**
> 
> ```
> await Get.generalDialog(pageBuilder: (context, __, ___) => AlertDialog());
> ```
> 
> P.S. Thanks for this awesome library! It makes separation of UI/logic effortless when paired up with other packages.

Hi, thanks for opening this issue. I think maybe this is not so well documenting, but Get.generalDialog is exactly a showGeneralDialog using the Get context.

The imposition of `!BarrierDismissible || barrierLabel != null` is in the Flutter API
https://api.flutter.dev/flutter/widgets/showGeneralDialog.html

I think you might be looking for:
Get.dialog(AlertDialog()); which is similar to the showDialog of the Flutter API.

**Comment by MaskyS** [comment]
From the doc comments of `showGeneralDialog`: 

> The barrierDismissible argument is used to determine whether this route can be dismissed by tapping the modal barrier. This argument defaults to true. If barrierDismissible is true, a non-null barrierLabel must be provided.
> 
> The barrierLabel argument is the semantic label used for a dismissible barrier. This argument defaults to "Dismiss".
> 
> The barrierColor argument is the color used for the modal barrier. This argument defaults to Color(0x80000000).
> 
> The transitionDuration argument is used to determine how long it takes for the route to arrive on or leave off the screen. This argument defaults to 200 milliseconds.

When using `generalDialog`, these default values are not applied- rather, they're being set to null when `Get` calls it internally- hence a dialog shown using as such: ```await Get.generalDialog(pageBuilder: (context, __, ___) => AlertDialog());``` will not act the same as ```await showGeneralDialog(context: context,  pageBuilder: (context, __, ___) => AlertDialog())```. In fact, the former will not even work.

This is what I hope will be fixed, as it contributes to a confusing/menial developer experience.

**Comment by MaskyS** [comment]
Also see: [routes.dart > _DialogRoute](https://github.com/flutter/flutter/blob/852a30b003a2b9eeefa12606d051c249b1991a82/packages/flutter/lib/src/widgets/routes.dart#L1639), whose constructor hosts the default values for the params I referred to above.

**Comment by jonataslaw** [comment]
> Also see: [routes.dart > _DialogRoute](https://github.com/flutter/flutter/blob/852a30b003a2b9eeefa12606d051c249b1991a82/packages/flutter/lib/src/widgets/routes.dart#L1639), whose constructor hosts the default values for the params I referred to above.

Hi, thanks for reporting this, I thought the final route would be something like
barrierDismissible ??= true,

But there is no nullity check, but an optional assignment in the constructor, which was being exchanged for null, because no value was being sent, you got it! Thank you!

Fix on version 2.5.7

**Comment by MaskyS** [comment]
@jonataslaw Please also add the default value for barrierLabel ("Dismiss"), as of now the generaldialog is still broken in 2.10.1

**Comment by jonataslaw** [comment]
Thanks again. Fixed in 2.10.2
